src/interactors/UnregisterDevice.js

```diff
@@ -1,15 +1,21 @@
 import logger from 'util/logger';
 
 class UnregisterDevice {
-  constructor(deviceStore, cloudConnector) {
+  constructor(deviceStore, cloudConnector, queue) {
```

Comment:
What do you think about receiving the parameter `queue` as `publisher` instead?

Reply:
Done

Reply:
Can you do this change for the other interactors too?

Reply:
Done